pkg/kubelet/cm/helpers.go

```diff
@@ -62,15 +63,26 @@ func buildContainerMapAndRunningSetFromRuntime(ctx context.Context, runtimeServi
 	runningSet := sets.New[string]()
 	containerMap := containermap.NewContainerMap()
 	containerList, _ := runtimeService.ListContainers(ctx, nil)
+	sort.SliceStable(containerList, func(i, j int) bool {
+		return containerList[i].CreatedAt < containerList[j].CreatedAt
+	})
 	for _, c := range containerList {
 		if _, exists := podSandboxMap[c.PodSandboxId]; !exists {
 			klog.InfoS("No PodSandBox found for the container", "podSandboxId", c.PodSandboxId, "containerName", c.Metadata.Name, "containerId", c.Id)
 			continue
 		}
 		podUID := podSandboxMap[c.PodSandboxId]
+		exist, err := containerMap.GetContainerID(podUID, c.Metadata.Name)
+		if err == nil {
+			// ContainerMap key is containerId, value is podUID&containerName.
+			// GetContainerID will iterator key-value pair, compair with value.
+			// We need delete the same value to avoid get unexpected containerID.
+			containerMap.RemoveByContainerID(exist)
```

Comment:
this is important enough (as the very issue we're fixing demonstrates) to warrant a log, perhaps at V(4)

Reply:
Added V(4) log, please take a look.